types/protoc-plugin/index.d.ts

```diff
@@ -0,0 +1,27 @@
+// Type definitions for protoc-plugin 0.0
+// Project: https://github.com/konsumer/node-protoc-plugin/
+// Definitions by: Jonny Reeves <https://github.com/jonnyreeves>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/// <reference types="node" />
+
+import * as stream from "stream";
+import {
+  CodeGeneratorRequest as pb_CodeGeneratorRequest,
+  CodeGeneratorResponse as pb_CodeGeneratorResponse
+} from "google-protobuf/google/protobuf/compiler/plugin_pb";
+import { FileDescriptorProto, SourceCodeInfo } from "google-protobuf/google/protobuf/descriptor_pb";
+import Location = SourceCodeInfo.Location;
+
+type OutputFiles = pb_CodeGeneratorResponse.File.AsObject[];
+type SimplePluginCallback = (filesToGenerate: ReadonlyArray<FileDescriptorProto.AsObject>) => (OutputFiles | Promise<OutputFiles>);
```

Comment:
If you want to import the type from the test, you can move them into the namespace

Reply:
I tried to add the `OutputFiles` type into the `namespace`, like so:

*index.d.ts*
```ts
declare namespace simplePlugin {
	type OutputFiles = pb_CodeGeneratorResponse.File.AsObject[];
	/* ... */
}
export = simplePlugin;
```

However when I try to import and use it in the tests like so, it fails:

*protoc-plugin-tests.ts*
```ts
import protocPlugin = require('protoc-plugin');
const {
  OutputFiles,
  /* ... */
} = protocPlugin;

protocPlugin(filesToGenerate => {
  const files: OutputFiles = [];
  /* ... */
});
```

However, when I run `npm run lint protoc-plugin`, I get the following error:

```
Error: /Users/jonny/Projects/DefinitelyTyped/types/protoc-plugin/protoc-plugin-tests.ts:7:3
ERROR: 7:3    expect  TypeScript@next compile error: 
Property 'OutputFiles' does not exist on type 'typeof simplePlugin'.
ERROR: 14:16  expect  TypeScript@next compile error: 
'OutputFiles' refers to a value, but is being used as a type here.
ERROR: 21:18  expect  TypeScript@next compile error: 
'OutputFiles' refers to a value, but is being used as a type here.
```

Reply:
Missing `export`: https://github.com/Microsoft/TypeScript-Handbook/blob/master/pages/declaration%20files/templates/module-function.d.ts.md

Reply:
When I try to export OutputFiles within the namespace, I get the following linting error:

```
ERROR: 21:2  strict-export-declare-modifiers  'export' keyword is redundant here because all declarations in this module are exported automatically. If you have a good reason to export some declarations and not others, add 'export {}' to the module to shut off automatic exporting. See: https://github.com/Microsoft/dtslint/blob/master/docs/strict-export-declare-modifiers.md
```